tests/helpers/test_debounce.py

```diff
@@ -41,6 +41,8 @@ async def test_immediate_works(hass: HomeAssistant) -> None:
     # Call and let timer run out
     await debouncer.async_call()
     assert len(calls) == 2
+    # Cancel the TimerHandle to avoid lingering tasks
+    debouncer._timer_task.cancel()
     await debouncer._handle_timer_finish()
```

Comment:
@MartinHjelmare / @balloob / @bdraco 
I'm pretty sure this is the root cause of the lingering task.

`debouncer._handle_timer_finish` is expected to run once the `self._timer_task` is finished.
Since the tests are calling `debouncer._handle_timer_finish` directly, the referenced task is set to `None` but never cancelled.

The flakyness is that sometimes hass is shutdown before the `call_later` is triggered (runs fine), and sometimes `call_later` gets triggered before hass shutdown (and task is lingering)